src/udev/udev-rules.c

```diff
@@ -365,24 +383,25 @@ UdevRules *udev_rules_free(UdevRules *rules) {
         return mfree(rules);
 }
 
-static int rule_resolve_user(UdevRules *rules, const char *name, uid_t *ret) {
+static int rule_resolve_user(UdevRuleLine *rule_line, const char *name, uid_t *ret) {
+        Hashmap **known_users = &LINE_GET_RULES(rule_line)->known_users;
```

Comment:
Missing `ASSERT_PTR`. The var. is dereferenced unconditionally later.

Reply:
Is it? `*known_users` is passed down to `hashmap_get` which is expected to handle NULL properly, isn't it?

Reply:
In this stage, `known_users` may be NULL, and will be allocated in this function.